### PR TITLE
[CORE] Drop redundant partial sort which has pre-project when offload sortAgg

### PR DESCRIPTION
**Pull request by zml1206**
## What changes were proposed in this pull request?
As titile.

## How was this patch tested?
UT

### CODE REVIEW

**Comment by zml1206** [comment]
Run Gluten Clickhouse CI

File: gluten-core/src/main/scala/org/apache/gluten/execution/SortUtils.scala [hunk 1]
```diff
@@ -23,11 +23,27 @@ import org.apache.spark.sql.execution.{ProjectExec, SortExec, SparkPlan}
 object SortUtils {
   def dropPartialSort(plan: SparkPlan): SparkPlan = plan match {
     case RewrittenNodeWall(p) => RewrittenNodeWall(dropPartialSort(p))
-    case sort: SortExec if !sort.global => sort.child
+    case PartialSortChild(p) => p
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/execution/SortUtils.scala:26]: Why we only drop partial sorts? I know it's previous code but I am still curious about that.

Also I'd suggest renaming:

`PartialSortChild(p)` -> `PartialSortLike(child)`
`ProjectChild(p)` -> `ProjectLike(child)`

Comment (zml1206) [reply gluten-core/src/main/scala/org/apache/gluten/execution/SortUtils.scala:26]: Sort generated by `EnsureRequirements`.
https://github.com/apache/spark/blob/1bfc9c3cbdc5d017fef68b0e84dd7fd22d2fef0f/sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala#L204

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala [hunk 2]
```diff
@@ -50,6 +50,32 @@ abstract class VeloxAggregateFunctionsSuite extends VeloxWholeStageTransformerSu
       .set("spark.sql.sources.useV1SourceList", "avro")
   }
 
+  test("drop redundant partial sort which has pre-project when offload sortAgg") {
```
Comment (zhztheplayer) [inline backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:53]: Would you like to move this test case down, to the bottom of this file, probably? Thanks.

**Review by zhztheplayer — APPROVED** [review]
Just some nits. Thank you.

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**